### DISCUSSION TITLE
增加通过android的TypeFace设置pag的字体

### DISCUSSION THREAD

**Discussion by zyc945** [Tencent/libpag#2217] [General] [answered]
android各种请求字体的方式最后都返回的是一个typeFace，这个拿不到路径，二进制数据也只能通过反射访问，有没有可能pag直接支持type去设置字体。

**Comment by kevingpqi123** [comment, marked as answer]
如果使用的是系统字体，获取到 TypeFace 的 fontFamily 和 fontStyle，传递给 PAGText 即可

**Reply by zyc945** [reply]
如果用google font是否无法直接使用

**Reply by kevingpqi123** [reply]
系统字体可以采用上面描述的方法，非系统字体直接传入字体路径，通过 PAGFont 的相关方法获取到 fontFamily 和 fontStyle

**Reply by zyc945** [reply]
google font一般是无法获取到字体路径的, 只能得到一个typeface实例.

**Reply by domchen** [reply]
获取不到路径的字体只要是系统目录里的，从typeface实例上查询名字，还是读取fontFamily和fontStyle，然后通过这个去创建PAGFont。
